tests/loader_test_vectors.rs

```diff
@@ -0,0 +1,231 @@
+{
+    let group_res: IonResult<Vec<_>> = raw_group
+        .iter()
+        .map(|elem| loader.load_all(elem.as_str().unwrap().as_bytes()))
+        .collect();
+    let group = group_res?;
+    for this in group.iter() {
+        for that in group.iter() {
+            group_assert(this, that);
+        }
+    }
+    Ok(())
+}
+
+/// Parses a document that has top-level `list`/`sexp` values that represent a *group*.
+/// If this top-level value is annotated with `embedded_documents`, then [`load_group_embedded`]
+/// is executed for that grouping.  Otherwise, the `value_assert` function is invoked for
+/// every pair of values in a group including the identity case (a value in a group is compared
+/// against itself).
+///
+/// Here is an example with two groups that might be used for equivalence testing:
+///
+/// ```plain
+/// (name $2 'name')
+/// embedded_documents::(
+///   "a name"
+///   "a $2"
+///   "'a' 'name'"
+/// )
+/// ```
+///
+/// This would have two groups, one with direct values that will be compared and another
+/// with embedded Ion text that will be parsed and compared.
+fn load_group<L, F1, F2>(
+    loader: L,
+    file_name: &str,
+    value_assert: F1,
+    group_assert: F2,
+) -> IonResult<()>
+where
+    L: Loader,
+    F1: Fn(&OwnedElement, &OwnedElement) -> (),
+    F2: Fn(&Vec<OwnedElement>, &Vec<OwnedElement>) -> (),
+{
+    let group_lists = load_file(&loader, file_name)?;
+    for group_list in group_lists.iter() {
+        // every grouping set is a list/sexp
+        // look for the embbedded annotation to parse/test as the underlying value
+        let is_embedded = group_list
+            .annotations()
+            .find(|annotation| annotation.text() == Some("embedded_documents"))
+            .is_some();
+        match group_list.as_sequence() {
+            Some(group) => {
+                if is_embedded {
+                    load_group_embedded(&loader, group, &group_assert)?;
+                } else {
+                    for this in group.iter() {
+                        for that in group.iter() {
+                            value_assert(this, that);
+                        }
+                    }
+                }
+            }
+            _ => return decoding_error(format!("Expected a sequence for group: {:?}", group_list)),
+        };
+    }
+    Ok(())
+}
+
+#[test_resources("ion-tests/iontestdata/good/equivs/**/*.ion")]
+#[test_resources("ion-tests/iontestdata/good/equivs/**/*.10n")]
+fn equivs(file_name: &str) {
+    let skip_list = concat(ALL_SKIP_LIST, EQUIVS_SKIP_LIST);
+    assert_file(&skip_list[..], file_name, || {
+        load_group(
+            loader(),
+            file_name,
+            |this, that| assert_eq!(this, that),
+            |this_group, that_group| assert_eq!(this_group, that_group),
+        )
+    });
+}
+
+#[test_resources("ion-tests/iontestdata/good/non-equivs/**/*.ion")]
+// no binary files exist and the macro doesn't like empty globs...
```

Comment:
Did you open an issue for this? The maintainer's response would be another good data point to help us decide whether it's worth it to fork the repo.

Reply:
Added frehberg/test-generator#12